src/reducer.js

```diff
@@ -334,12 +334,17 @@ const onFetchHistorySuccess = (state, action) => {
       }
       // add this tx to the history of the corresponding token
       tokenHistory.push(getTxHistoryFromTx(tx, tokenUid, tokenTxBalance));
-      const totalBalance = getBalance(tokenUid);
-      // update token total balance
-      tokensBalance[tokenUid] = totalBalance;
+      tokenUidSeen[tokenUid] = 1;
     }
   }
 
+  const tokensBalance = {};
+  for (const tokenUid in tokenUidSeen) {
```

Comment:
Do we need `tokenUidSeen`? Isn't it the same as the keys of `tokenHistory`?

If we do need, maybe a `set` is better than a `dict`.

Reply:
You're right; we can use `tokensHistory`. I haven't used it before because we don't have to update the balance of all tokens, but it always start empty, so we can safely use it. Can you review again, please?